hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/MySQLUpsertQueryBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.jdbc;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+class MySQLUpsertQueryBuilder {
+
+    private final String query;
+
+    MySQLUpsertQueryBuilder(JdbcTable jdbcTable) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        getInsertClause(jdbcTable, stringBuilder);
+        getValuesClause(jdbcTable, stringBuilder);
+        getOnDuplicateClause(jdbcTable, stringBuilder);
+
+        query = stringBuilder.toString();
+    }
+
+    protected void getInsertClause(JdbcTable jdbcTable, StringBuilder stringBuilder) {
+
+        stringBuilder.append("INSERT INTO ")
+                .append(jdbcTable.getExternalName())
+                .append(" (")
+                .append(String.join(",", jdbcTable.dbFieldNames()))
+                .append(") ");
+    }
+
+    protected void getValuesClause(JdbcTable jdbcTable, StringBuilder stringBuilder) {
+
+        List<String> dbFieldNames = jdbcTable.dbFieldNames();
+
+        String values = dbFieldNames.stream()
+                .map(dbFieldName -> "?")
+                .collect(Collectors.joining(","));
```

Comment:
They are not the same. 
The first one is giving "?,?,?" string . 
The suggested one is giving "field1,field2,field3" string . 
We need to use ? character instead of field name

Reply:
sorry, my bad :)